pkg/clients/gcp.go

```diff
@@ -95,9 +97,24 @@ func UseProviderConfig(ctx context.Context, c client.Client, mg resource.Managed
 		if err != nil {
 			return "", nil, errors.Wrap(err, "cannot get credentials")
 		}
-		return pc.Spec.ProjectID, option.WithCredentialsJSON(data), nil
+		if isJSON(data) {
+			return pc.Spec.ProjectID, option.WithCredentialsJSON(data), nil
+		} else {
+			t := oauth2.Token{
+				AccessToken: string(data),
+			}
+			if ok := t.Valid(); !ok {
+				return pc.Spec.ProjectID, opts, errors.New("Access token invalid")
+			}
+			ts := oauth2.StaticTokenSource(&t)
+			return pc.Spec.ProjectID, option.WithTokenSource(ts), nil
+		}
```

Comment:
You can drop this `else` and outdent this block:

```suggestion
		if isJSON(data) {
			return pc.Spec.ProjectID, option.WithCredentialsJSON(data), nil
		}
		t := oauth2.Token{
			AccessToken: string(data),
		}
		if ok := t.Valid(); !ok {
			return pc.Spec.ProjectID, opts, errors.New("access token invalid")
		}
		ts := oauth2.StaticTokenSource(&t)
		return pc.Spec.ProjectID, option.WithTokenSource(ts), nil
```

Reply:
Fixed